Add sensor for Lyft time and price (based on Uber sensor)

drkp opened the pull request:
## Description:

We should have a sensor for Lyft ETA and prices, since we have one for Uber. 

**Related issue (if applicable):** n/a

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** forthcoming

## Example entry for `configuration.yaml` (if applicable):
```
sensor:
  - platform: lyft
    client_id: [...]
    client_secret: [...]
    start_latitude: 37.8116380 
    start_longitude: -122.2648050
    end_latitude: 37.615223
    end_longitude: -122.389977
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

drkp committed:
Add sensor for Lyft time and price (based on Uber sensor)

homeassistant commented:
Hi @drkp, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#6711).

Once you do that we will be able to review and accept this pull request.

Thanks!

balloob commented on homeassistant/components/sensor/lyft.py:
```diff
@@ -0,0 +1,231 @@
+    vol.Required(CONF_CLIENT_SECRET): cv.string,
+    vol.Required(CONF_START_LATITUDE): cv.latitude,
+    vol.Required(CONF_START_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_END_LATITUDE): cv.latitude,
```
These are marked optional but I don't see any code that assumes it's not there?

drkp replied:
We check if the value is None in line 215.

balloob commented on homeassistant/components/sensor/lyft.py:
```diff
@@ -0,0 +1,231 @@
+            if estimate is not None:
+                self._unit_of_measurement = estimate.get('currency')
+
+        self.update()
```
Please don't call `self.update()` but instead pass `True` as second parameter to `add_devices`

balloob commented on homeassistant/components/sensor/lyft.py:
```diff
@@ -0,0 +1,231 @@
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if 'lyft' not in self._name.lower():
```
Please do this check in the constructor.

drkp committed:
Minor fixes to lyft sensor
 - use add_devices(...,True) instead of explicitly calling update
 - move sensor name check into constructor

drkp committed:
lyft sensor: disable sandbox mode

drkp commented:
Made the requested changes. Also noticed that the lyft_rides module defaults to sandbox (vs production) mode, so disabled that.

balloob commented:
Don't forget a docs PR!

drkp commented:
Already there, it's home-assistant/home-assistant.github.io#2304 :-)
